consensus/consortium/v2/consortium.go

```diff
@@ -358,6 +331,33 @@ func (c *Consortium) verifyCascadingFields(chain consensus.ChainHeaderReader, he
 		return consensus.ErrUnknownAncestor
 	}
 
+	// Check extra data
+	isShillin := c.chainConfig.IsShillin(header.Number)
+	extraData, err := finality.DecodeExtra(header.Extra, isShillin)
+	if err != nil {
+		return err
+	}
+
+	// Check extra data
+	isEpoch := number%c.config.EpochV2 == 0 || c.chainConfig.IsOnConsortiumV2(header.Number)
+
+	if !isEpoch && len(extraData.CheckpointValidators) != 0 {
+		return consortiumCommon.ErrExtraValidators
+	}
+
+	if isShillin && extraData.HasFinalityVote == 1 {
+		if err := c.verifyFinalitySignatures(
```

Comment:
For me, it looks good to move this verification here; yet we call `snapshot` twice including one in `verifyCascadingFields` one in line 376, should we better this?

Reply:
This could be implemented in another PR